contracts/linear/src/validator_pool.rs

```diff
@@ -1000,8 +1017,12 @@ impl Validator {
         pool.save_validator(self);
     }
 
-    pub fn sync_account_balance(&mut self, pool: &mut ValidatorPool) -> Promise {
-        self.pre_execution(pool);
+    /// Due to shares calculation and rounding of staking pool contract,
+    /// the amount of staked and unstaked balance might be a little bit
+    /// different than we requested.
+    /// This method is to sync the actual numbers with the validator.
+    pub fn sync_account_balance(&mut self) -> Promise {
+        require!(self.executing, ERR_VALIDATOR_SYNC_WHEN_UNLOCKED);
```

Comment:
What about `return ;` instead of panic if `executing` is `true`? 

Just curious. Why panic is preferred here?

Reply:
Because we don't expect it to be called when unlocked in current scenes? It's a defensive programming here IMO.